docker/backup.postgres.plugin

```diff
@@ -89,6 +90,19 @@ function onRestoreDatabase(){
       gunzip -c "${_fileName}" | psql -v ON_ERROR_STOP=1 -x -h "${_hostname}" ${_portArg} -d "${_database}"
       # Get the status code from psql specifically.  ${?} would only provide the status of the last command, psql in this case.
       _rtnCd=${PIPESTATUS[1]}
+      IFS=\0;
```

Comment:
Is there an easier way to do this?  It feels overly complicated.

Reply:
The easier/cleaner way would be to switch over to using `pg_restore`. I am a fan of switching over to that, but I thought you had some reservations. Changes would at that point need to include:
- unzipping backup file to a `temp` location
- run `pg_restore` with the appropriate flags
- clean up
I think it is doable, but will require some additional testing as the restore process will likely have a couple of differences.

Reply:
I was wondering, more, if there was an more concise way of restoring the permissions on the database with this current approach.

If we were to introduce `pg_restore` based backup and restore capabilities, I'd want to introduce it as an option.  Likely as an add-on plug-in that would be loaded based on a configuration option, and provide completely alternate implementations for `onBackupDatabase` and `onRestoreDatabase`.